### PR TITLE
feat(css/minifier): Handle nested `calc()`

### PR DESCRIPTION
**Pull request by ghostd**
**Description:**

Handle nested `calc()`:
`calc(100% - calc(50% + 25px))` => `calc(50% + 25px)`

**BREAKING CHANGE:**

**Related issue (if exists):**

### CODE REVIEW

File: crates/swc_css_minifier/tests/fixture/compress-calc/discard-zero/output.min.css [hunk 1]
```diff
@@ -1 +1 @@
-.class1{width:calc(100vw/2 - 6px)}.class2{width:500px}.class3{width:calc(0 - 10px)}.class4{width:calc(0 - calc(1px + 1em))}.class5{width:calc(0 - (100vw - 10px)/2)}.class6{width:calc(10px - 100vw)}.class7{width:calc(0px - (100vw - 10px)*2)}.class8{width:calc(-10px - 100vw)}.class9{width:calc(0px - var(--foo,4px)/2)}
+.class1{width:calc(100vw/2 - 6px)}.class2{width:500px}.class3{width:calc(0 - 10px)}.class4{width:calc(0 - 1px - 1em)}.class5{width:calc(0 - (100vw - 10px)/2)}.class6{width:calc(10px - 100vw)}.class7{width:calc(0px - (100vw - 10px)*2)}.class8{width:calc(-10px - 100vw)}.class9{width:calc(0px - var(--foo,4px)/2)}
```
Comment (alexander-akait) [inline crates/swc_css_minifier/tests/fixture/compress-calc/discard-zero/output.min.css:1]: `width:calc(0 - 1px - 1em)` -> `width:calc(-1px - 1em)`

**Review by alexander-akait — CHANGES_REQUESTED** [review]
nested `calc()` is like (` and `)` , so I think we should handle it not in `transform_calc_value_into_component_value`, we should handle them inside calculation.

**Comment by ghostd** [comment]
@alexander-akait 'im not sure to understand your comment, i added the "nested calc" handling into the `resume` function. I rebased, i hope the diff will be more readable.

**Review by alexander-akait — APPROVED** [review]
Great, there are some impvomenets more, like `width:calc(2.25rem - 0px)`, but let's do it another PR, because it is not related to nested `calc`

**Comment by alexander-akait** [comment]
@kdy1 Ready to review

**Review by kdy1 — APPROVED** [review]
Thanks!

---

swc-bump:
 - swc_css_minifier

**Comment by ghostd** [comment]
@alexander-akait i'll work on removing zeroes this week.

**Comment by alexander-akait** [comment]
Thanks

**Pull request merged**